src/basic/string-util.c

```diff
@@ -295,6 +295,49 @@ static int write_ellipsis(char *buf, bool unicode) {
         return 3;
 }
 
+static size_t ansi_sequence_length(const char *s) {
+        if (s[0] != 0x1B)
+                return 0;  /* Not the start of a sequence */
+
+        if (s[1] == 0x5B) { /* CSI sequence */
+                size_t i = 2;
+
+                while (s[i] && s[i] >= 0x30 && s[i] <= 0x3F) /* Parameter bytes */
+                        i++;
+                while (s[i] && s[i] >= 0x20 && s[i] <= 0x2F) /* Intermediate bytes */
+                        i++;
+                if (s[i] < 0x40 || s[i] > 0x7E) /* Final byte */
+                        return 0;  /* Bad sequence */
+                return i + 1;
+        } else if (s[1] >= 0x40 && s[1] <= 0x5F) /* Fe sequence. */
+                return 2;
+
+        return 0;  /* Bad escape? */
+}
+
+static bool string_has_ansi_sequence(const char *s, size_t len) {
+        for (const char *t = s; t < s + len; t++)
```

Comment:
i figure performance reasons it would probably a ton better to do:

```c
for (const t = memchr(s, 0x1B, len); t; len -= t - s + 1, s = t + 1) {
        if (ansi_sequence_length(t) > 0)
                return true;
}
```

Reply:
btw, this will result in bad memory access if called with len <= 1, and even otherwise, if 0x1B is the last char in the string...